### PR TITLE
[quant] Support custom qmin/qmax for activation and weight for xnnpack quantizer

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117305

Summary:
att, this allows us to experiment with 4 bit quant in xnnpack

Test Plan:
python test/test_quantization.py -k test_dynamic_linear_int4_weight

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/quantization/pt2e/test_xnnpack_quantizer.py [hunk 1]
```diff
@@ -534,6 +534,54 @@ def test_dynamic_linear(self):
...
+            weight=per_channel_weight_observer_range_neg_127_to_127.with_args(
+                quant_min=0, quant_max=15
+            ),
+        )
+        qconfig_mapping = QConfigMapping().set_global(qconfig)
+        # Test with 2d inputs
+        example_inputs_2d = (torch.randn(9, 8),)
+        example_inputs_4d = (torch.randn(9, 10, 11, 8),)
+        for example_inputs in [example_inputs_2d, example_inputs_4d]:
+            self._test_quantizer(
```
Comment (digantdesai) [inline test/quantization/pt2e/test_xnnpack_quantizer.py:575]: Does this check the value ranges on the q/dq nodes?

**Review by digantdesai — APPROVED** [review]
Looks good to me, thanks Jerry.

File: test/quantization/pt2e/test_xnnpack_quantizer.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_xnnpack_quantizer.py:575]: yes, this checks the end result from pt2e flow and fx flow are the same. I also tried modifying the ranges and it failed

**Pull request closed without merging**